tests/unit/template_cassdc_test.go

```diff
@@ -114,6 +114,19 @@ var _ = Describe("Verify CassandraDatacenter template", func() {
 			Expect(config.JvmOptions.InitialHeapSize).To(BeEmpty())
 			Expect(config.JvmOptions.MaxHeapSize).To(BeEmpty())
 			Expect(config.JvmOptions.YoungGenSize).To(BeEmpty())
+
+			// Verify (writeable) volumes
+			Expect(cassdc.Spec.PodTemplateSpec.Spec.Containers[0].VolumeMounts[0].Name).To(Equal("cassandra-config"))
```

Comment:
I could be wrong, but I assume that the order of the volume mounts does not matter. If this assumption is valid could you change these calls to not rely on ordering? It will make the tests less brittle.

See [here](https://github.com/k8ssandra/k8ssandra/blob/main/tests/unit/template_cassdc_test.go#L499) for a related example. It verifies the containers without relying on order.

Reply:
I agree. These are the first lines of Go I've written, so I must admit to taking the easy way out. I'll look at your example. I also wanted to add a unit test for verifying that all the names of volumeMounts for all the containers match a name in the volumes list, but again I'm don't know the idiomatic Go way of doing it, but I'll give it a go.